src/main/java/net/dries007/tfc/objects/blocks/agriculture/BlockFruitTreeTrunk.java

```diff
@@ -0,0 +1,339 @@
+                                }
+                            }
+                            else
+                            {
+                                //Time to grow one trunk
+                                worldIn.setBlockState(pos.up(), this.getDefaultState());
+                                //Set all remaining leaf blocks to air
+                                levelUpLeaves(worldIn, pos, false);
+                            }
+                        }
+                        else
+                        {
+                            boolean flag = false; //If true, just grown a leaf/branch
+                            if (worldIn.getBlockState(pos.up()).getBlock() != BlockFruitTreeBranch.get(tree))
+                            {
+                                BlockPos missingLeaf = getMissingLeaf(worldIn, pos);
+                                if (missingLeaf != null)
+                                {
+                                    //There is a missing leaf
+                                    flag = true;
+                                    if (worldIn.getBlockState(missingLeaf).getMaterial().isReplaceable())
+                                    {
+                                        worldIn.setBlockState(missingLeaf, BlockFruitTreeLeaves.get(tree).getDefaultState().withProperty(BlockFruitTreeLeaves.HARVESTABLE, true));
+                                    }
+                                }
+                                else
+                                {
+                                    flag = true;
+                                    worldIn.setBlockState(pos.up(), BlockFruitTreeBranch.get(tree).getDefaultState());
+                                    levelUpLeaves(worldIn, pos, true);
+                                }
+                            }
+                            else
+                            {
+                                BlockPos missingLeaf = getMissingLeaf(worldIn, pos.up()); //Please note that this is pos.up()
+                                if (missingLeaf != null)
+                                {
+                                    //There is a missing leaf
+                                    flag = true;
+                                    if (worldIn.getBlockState(missingLeaf).getMaterial().isReplaceable())
+                                    {
+                                        worldIn.setBlockState(missingLeaf, BlockFruitTreeLeaves.get(tree).getDefaultState().withProperty(BlockFruitTreeLeaves.HARVESTABLE, true));
+                                    }
+                                }
+                            }
+                            //Starting with top
+                            for (EnumFacing branchFacing : EnumFacing.VALUES)
+                            {
+                                if (flag) break;
```

Comment:
Is there any reason that this loop needs all the `if (flag) break;` instead of just using `break` where you set the flag to true?

Reply:
You're right, i could just change `te.resetCounter()` and use return in the inner loop.